huaweicloud/services/acceptance/apig/data_source_huaweicloud_apig_channel_member_groups_test.go

```diff
@@ -0,0 +1,128 @@
+package apig
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDataSourceChannelMemberGroups_basic(t *testing.T) {
+	var (
+		rName = acceptance.RandomAccResourceName()
+
+		dataSource = "data.huaweicloud_apig_channel_member_groups.test"
+		dc         = acceptance.InitDataSourceCheck(dataSource)
+
+		byName   = "data.huaweicloud_apig_channel_member_groups.filter_by_name"
+		dcByName = acceptance.InitDataSourceCheck(byName)
+
+		byExactName   = "data.huaweicloud_apig_channel_member_groups.filter_by_exact_name"
+		dcByExactName = acceptance.InitDataSourceCheck(byExactName)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckApigSubResourcesRelatedInfo(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceChannelMemberGroups_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestMatchResourceAttr(dataSource, "member_groups.#", regexp.MustCompile(`^[1-9]([0-9]*)?$`)),
+					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.name"),
+					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.create_time"),
+					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.update_time"),
```

Comment:
```suggestion
					resource.TestMatchResourceAttr(dataSource, "member_groups.#", regexp.MustCompile(`^[1-9]([0-9]*)?$`)),
					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.id"),
					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.name"),
					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.create_time"),
					resource.TestCheckResourceAttrSet(dataSource, "member_groups.0.update_time"),
```
Check other properties.

Reply:
done